.github/workflows/deploy-env.yaml

```diff
@@ -0,0 +1,51 @@
+name: Deploy Environment
+
+on:
+  push:
+  workflow_dispatch:
+    inputs:
+      environment:
+        description: 'Environment to deploy to'
+        type: choice
+        required: true
+        default: 'ops-preview-1'
+        options:
+        - ALL
+        - LIVE
+        - live-sanchonet@eu-central-1@v2
+        - DEV
+        - live-mainnet@eu-central-1@v2
+        - live-sanchonet@us-east-2@v1
+        - dev-mainnet@us-east-1
+        - live-mainnet@us-east-2@v2
+        - live-sanchonet@us-east-2@v2
+        - dev-preprod@us-east-1@v2
+        - live-preprod@eu-central-1@v2
+        - OPS
+        - dev-preview@us-east-1
+        - live-preprod@us-east-2@v2
+        - ops-preprod-1@us-east-1
+        - dev-sanchonet@us-east-1@v1
+        - live-preview@eu-central-1@v2
+        - ops-preview-1@us-east-1
+        - dev-sanchonet@us-east-1@v2
+        - live-preview@us-east-2@v2
+        - staging-preprod@us-east-1@v2
+
+jobs:
+  deploy:
+    environment: ${{ inputs.environment || 'ops-preview-1@us-east-1' }}
+    runs-on: ubuntu-latest
+    steps:
+      - name: 📥 Checkout repository
+        uses: actions/checkout@v3
+
+      - name: 🧰 Setup Nix
+        uses: cachix/install-nix-action@v21
+
+      - name: 🚀 Deploy
+        run: |
+          echo "${{ secrets.ENVRC }}" > .envrc.local
+          source .envrc.local
+          nix develop
+          echo "yes" | nix run .#cardano-services.${{ inputs.environment || 'ops-preview-1@us-east-1' }}.apply
```

Comment:
I don't know how I feel about having this default? Can you say more?

Reply:
Ah, silly me, I only read the code, not your comments:

> I am going to take out the `push:` trigger and the `ops-preview-1@us-east-1` default value in instances of `{{ inputs.environment || 'ops-preview-1@us-east-1' }}`.

Reply:
I already addressed that here: https://github.com/input-output-hk/cardano-js-sdk/pull/1382#issuecomment-2249679228. The default values are to test the workflow. The approach is to supply this value using a `workflow_dispatch` interface which is not currently available for use since this workflow is not on `master`.

I would take out the default values before merge once the PR is approved.

Reply:
Sillier me, I had the page already opened down and it did not update with your most recent comment. Kindly ignore my reply. 😅